src/google-maps/google-map/google-map.ts

```diff
@@ -17,28 +34,93 @@ import {ReplaySubject} from 'rxjs';
   changeDetection: ChangeDetectionStrategy.OnPush,
   template: '<div class="map-container"></div>',
 })
-export class GoogleMap implements OnInit {
-  // Arbitrarily chosen default size
-  @Input() height = '500px';
-  @Input() width = '500px';
+export class GoogleMap implements OnInit, OnDestroy {
+  @Input()
+  set height(height: string) {
+    this._height$.next(height || DEFAULT_HEIGHT);
+  }
 
-  // TODO(mbehrlich): add options, handlers, properties, and methods.
+  @Input()
+  set width(width: string) {
+    this._width$.next(width || DEFAULT_WIDTH);
+  }
 
-  private readonly _map$ = new ReplaySubject<google.maps.Map>(1);
+  @Input()
+  set center(center: google.maps.LatLngLiteral) {
+    this._center$.next(center);
+  }
+  @Input()
+  set zoom(zoom: number) {
+    this._zoom$.next(zoom);
+  }
+  @Input()
+  set options(options: google.maps.MapOptions) {
+    this._options$.next(options || DEFAULT_OPTIONS);
+  }
 
-  constructor(private readonly _elementRef: ElementRef) {}
+  // TODO(mbehrlich): Add event handlers, properties, and methods.
 
-  ngOnInit() {
-    // default options set to the Googleplex
-    const options: google.maps.MapOptions = {
-      center: {lat: 37.421995, lng: -122.084092},
-      zoom: 17,
-    };
+  private readonly _height$ = new BehaviorSubject<string>(DEFAULT_HEIGHT);
+  private readonly _width$ = new BehaviorSubject<string>(DEFAULT_WIDTH);
+  private readonly _options$ = new BehaviorSubject<google.maps.MapOptions>(DEFAULT_OPTIONS);
+  private readonly _center$ = new BehaviorSubject<google.maps.LatLngLiteral|undefined>(undefined);
+  private readonly _zoom$ = new BehaviorSubject<number|undefined>(undefined);
 
+  private readonly _destroy$ = new Subject<void>();
+
+  constructor(private readonly _elementRef: ElementRef) {
+    const googleMapsWindow: GoogleMapsWindow = window;
+    if (!googleMapsWindow.google) {
+      throw new Error(
```

Comment:
You can omit `new`; `throw Error(...)` functions the same way

(here and elsewhere)

Reply:
Done.